tests/std/tests/P2321R2_views_zip_transform/test.cpp

```diff
@@ -0,0 +1,864 @@
+                    == is_noexcept);
+            }
+
+            // ... with const rvalue arguments
+            STATIC_ASSERT(CanZipTransform<TransformType, const remove_reference_t<RangeTypes>...>
+                          == (are_views && (copy_constructible<AllView<RangeTypes>> && ...)
+                              && move_constructible<TransformType>) );
+            if constexpr (CanZipTransform<TransformType, const remove_reference_t<RangeTypes>...>) {
+                using ExpectedZipTransformType =
+                    ranges::zip_transform_view<TransformType, AllView<const remove_reference_t<RangeTypes>>...>;
+                constexpr bool is_noexcept = is_nothrow_move_constructible_v<ranges::_Movable_box<TransformType>>
+                                          && is_nothrow_constructible_v<InnerView, const AllView<RangeTypes>&&...>;
+
+                STATIC_ASSERT(same_as<decltype(views::zip_transform(std::forward<TransformType_>(transform),
+                                          std::move(as_const(test_ranges))...)),
+                    ExpectedZipTransformType>);
+                STATIC_ASSERT(noexcept(views::zip_transform(
+                                  std::forward<TransformType_>(transform), std::move(as_const(test_ranges))...))
+                              == is_noexcept);
+            }
+        }
+
+        if constexpr (move_constructible<TransformType>) {
+            // Validate deduction guide
+            same_as<ZipTransformType> auto zipped_transformed_range = ranges::zip_transform_view{
+                std::forward<TransformType_>(transform), std::forward<RangeTypes>(test_ranges)...};
+
+            // Validate zip_transform_view::size()
+            STATIC_ASSERT(CanMemberSize<ZipTransformType> == ranges::sized_range<InnerView>);
+            if constexpr (CanMemberSize<ZipTransformType>) {
+                using expected_size_type                            = decltype(declval<InnerView&>().size());
+                same_as<expected_size_type> auto zip_transform_size = zipped_transformed_range.size();
+
+                assert(zip_transform_size == ranges::size(transformed_elements));
+                STATIC_ASSERT(noexcept(zipped_transformed_range.size()) == noexcept(declval<InnerView&>().size()));
+            }
+
+            STATIC_ASSERT(CanMemberSize<const ZipTransformType> == ranges::sized_range<const InnerView>);
+            if constexpr (CanMemberSize<const ZipTransformType>) {
+                using expected_size_type                            = decltype(declval<const InnerView&>().size());
+                same_as<expected_size_type> auto zip_transform_size = as_const(zipped_transformed_range).size();
+
+                assert(zip_transform_size == ranges::size(transformed_elements));
+                STATIC_ASSERT(
+                    noexcept(zipped_transformed_range.size()) == noexcept(declval<const InnerView&>().size()));

```

Comment:
The LHS omitted `as_const` here; it should be `as_const(zipped_transformed_range).size()`.